e2e/playground.py

```diff
@@ -0,0 +1,2135 @@
+
+
+class _OutputImageBase(_OutputInlineContainer, _OutputBase):
+    # id: str
+    # width: str = "100%"
+    # height: str = "400px"
+    # inline: bool = False
+
+    def __init__(self, page: Page, id: str, loc_classes: str = "") -> None:
+        super().__init__(
+            page,
+            id=id,
+            loc=f"#{id}.shiny-image-output{loc_classes}",
+        )
+        self.loc_img = self.loc.locator("img")
+
+    loc_img: Locator
+
+    def expect_height_to_have_value(
+        self,
+        value: AttrValue,
+        *,
+        timeout: Timeout = None,
+    ) -> None:
+        self.expect_inline(inline=value is None, timeout=timeout)
+        expect_el_style(self.loc, "height", value, timeout=timeout)
+
+    def expect_width_to_have_value(
+        self,
+        value: AttrValue,
+        *,
+        timeout: Timeout = None,
+    ) -> None:
+        self.expect_inline(inline=value is None, timeout=timeout)
+        expect_el_style(self.loc, "width", value, timeout=timeout)
+
+    def expect_img_src(
+        self,
+        value: AttrValue,
+        *,
+        timeout: Timeout = None,
+    ) -> None:
+        expect_attr(self.loc_img, "src", value, timeout=timeout)
+
+    def expect_img_width(
+        self,
+        value: AttrValue,
+        *,
+        timeout: Timeout = None,
+    ) -> None:
+        expect_attr(self.loc_img, "width", value, timeout=timeout)
+
+    def expect_img_height(
+        self,
+        value: AttrValue,
+        *,
+        timeout: Timeout = None,
+    ) -> None:
+        expect_attr(self.loc_img, "height", value, timeout=timeout)
+
+    def expect_img_alt(
+        self,
+        value: AttrValue,
+        *,
+        timeout: Timeout = None,
+    ) -> None:
+        expect_attr(self.loc_img, "alt", value, timeout=timeout)
+
+    def expect_img_style(
+        self,
+        value: AttrValue,
+        *,
+        timeout: Timeout = None,
+    ) -> None:
+        expect_attr(self.loc_img, "style", value, timeout=timeout)
+
+
+class OutputImage(_OutputImageBase):
+    def __init__(self, page: Page, id: str) -> None:
+        super().__init__(page, id=id)
+        self.loc_img = self.loc.locator("img")
+
+
+class OutputPlot(_OutputImageBase):
+    # shiny-plot-output
+    # id: str
+    # width: str = "100%"
+    # height: str = "400px"
+    # inline: bool = False
+    def __init__(self, page: Page, id: str) -> None:
+        super().__init__(page, id=id, loc_classes=".shiny-plot-output")
+
+
+class OutputUi(_OutputContainer, _OutputBase):
+    # id: str,
+    # inline: bool = False,
+    # container: Optional[TagFunction] = None,
+    def __init__(self, page: Page, id: str) -> None:
+        super().__init__(page, id=id, loc=f"#{id}")
+
+    # TODO-barret; Should we do `expect_html_to_have_value()`?
+    # Thinking they can call `expect(self.loc).to_have_html(value)` directly as Shiny does not own that value, the user does.
```

Comment:
How about a method that asserts that it's not empty? (and succeeded)

Reply:
64d5694